Client/Coordinators/Library/LibraryCoordinator.swift

```diff
@@ -32,10 +36,38 @@ class LibraryCoordinator: BaseCoordinator, LibraryPanelDelegate {
     func start(with homepanelSection: Route.HomepanelSection) {
         libraryViewController.setupOpenPanel(panelType: homepanelSection.libraryPanel)
         libraryViewController.delegate = self
+        libraryViewController.navigationHandler = self
         libraryViewController.resetHistoryPanelPagination()
     }
 
-    // MARK: - LibraryCoordinatorDelegate
+    // MARK: - LibraryNavigationHandler
+
+    func start(panelType: LibraryPanelType, navigationController: UINavigationController) {
+        switch panelType {
+        case .bookmarks:
+            createBookmarksCoordinator(navigationController: navigationController)
+        case .history:
+            break
+        case .downloads:
+            break
+        case .readingList:
+            break
+        }
+    }
+
+    private func createBookmarksCoordinator(navigationController: UINavigationController) {
+        guard !childCoordinators.contains(where: { $0 is BookmarksCoordinator }) else { return }
+        let router = DefaultRouter(navigationController: navigationController)
+        let bookmarksCoordinator = BookmarksCoordinator(
+            router: router,
+            profile: profile,
+            parentCoordinator: parentCoordinator
+        )
+        add(child: bookmarksCoordinator)
+        (navigationController.topViewController as? BookmarksPanel)?.bookmarkCoordinatorDelegate = bookmarksCoordinator
```

Comment:
Can we access this another way than from the `navigationController.topViewController`?

Reply:
i honestly don't know another way now, i will think on it

Reply:
Looking at the code, it seems we already set the `BookmarksPanel` from the `BookmarksCoordinator` with `controller.bookmarkCoordinatorDelegate = self`. I think that enables us to remove that line here?

Reply:
Okkk so turns out, this is needed due to `LibraryPanelDescriptor` creating the first `BookmarksPanel` from the `LibraryPanelHelper` > `LibraryViewModel` > `LibraryViewController`. This goes out of scope of this PR, but we'll have some conversations to determine if we want to refactor those `LibraryPanelDescriptor`. They doesn't seems to align much with the coordinator pattern, which means a refactor of those. If we go that route, this means another set of tasks - so let's continue with this PR for now as is.

Reply:
FYI I created a task for this https://github.com/mozilla-mobile/firefox-ios/issues/15707